scripts/entrypoint.sh

```diff
@@ -29,6 +29,17 @@ mkdir -p  "${GEOSERVER_DATA_DIR}" "${CERT_DIR}" "${FOOTPRINTS_DATA_DIR}" "${FONT
 source /scripts/functions.sh
 source /scripts/env-data.sh
 
+# Rename to match wanted context-root and so that we can unzip plugins to
+# existing directory.
+if [ x"${GEOSERVER_CONTEXT_ROOT}" != xgeoserver ]; then
+  echo "INFO: changing context-root to '${GEOSERVER_CONTEXT_ROOT}'."
+  if [ -e "${CATALINA_HOME}/webapps/geoserver" ]; then
```

Comment:
The webapp is not only deployed as `"${CATALINA_HOME}/webapps/geoserver"`, in some cases it can be defined as https://github.com/kartoza/docker-geoserver/blob/develop/scripts/setup.sh#L97

Reply:
I think maybe the case in https://github.com/kartoza/docker-geoserver/blob/a54266835d1ec330f0f2920ce28839732c554ccd/scripts/setup.sh#L97-L102 probably do not have to be handled?

At build time, we'll just keep the default "geoserver" name.  It's only at run time that we need to rename it to match required context-root.

Unless I missed something, `setup.sh` is used at build time so we can just keep it as-is.  `entrypoint.sh` is used at run time so it has to change.

In fact, at build time, we have to use the default "geoserver" name so that at run time we know for sure from name to rename to the required context-root.

Reply:
Yes, the setup is run at build time but https://github.com/kartoza/docker-geoserver/blob/develop/scripts/entrypoint.sh#L102 
shows that if you install the platform-independent binary you can end up with your webapps as `/geoserver/webapps/geoserver`, so that would also need to be handled

Reply:
Oh thanks for pointing this out!  It seems we do not use this scenario so it never broke for us.  Will try to send a fix.

Reply:
I am confused, `GEOSERVER_HOME=/geoserver` so `${GEOSERVER_HOME}/start.jar` is `/geoserver/start.jar` but that file is not in the docker image.  So how do I get that file so I can test for this scenario?

Reply:
I think I finally understand what you meant.

There are 2 possible install dirs (CATALINA_HOME and GEOSERVER_HOME).  I already handed CATALINA_HOME, have to also handle GEOSERVER_HOME.

Hopefully this commit 6252ae3f3dc534d4bcf27a41ad0c11907ebfc936 is doing what you want.